mleap-runtime/src/main/scala/ml/combust/mleap/bundle/ops/feature/OneHotEncoderOp.scala

```diff
@@ -30,18 +30,22 @@ class OneHotEncoderOp extends MleapOp[OneHotEncoder, OneHotEncoderModel] {
 
     override def load(model: Model)
                      (implicit context: BundleContext[MleapContext]): OneHotEncoderModel = {
+      val dropLast = model.value("drop_last").getBoolean
       if (model.getValue("size").nonEmpty) {
         // Old version of 1HE.
-        OneHotEncoderModel(categorySizes = Array(model.value("size").getLong.toInt),
-          // assume HandleInvalid.Keep always since old version of 1HE didn't have a way to specify this
-          handleInvalid = HandleInvalid.Keep,
-          dropLast = model.value("drop_last").getBoolean)
+        OneHotEncoderModel(
+          categorySizes = Array(model.value("size").getLong.toInt),
+          handleInvalid =
+            model.getValue("handle_invalid").map(_.getString).map(HandleInvalid.fromString(_, false))
+              //this maintains backwards compatibility for models that don't have handle_invalid serialized
+              .getOrElse(if (dropLast) HandleInvalid.Keep else HandleInvalid.default),
```

Comment:
Should we make this `.getOrElse(HandleInvalid.Keep)` to match the hard-coded `handleInvalid = HandleInvalid.Keep` from before? 

Let's say there's an old bundle that had `drop_last = False` in it. If they were to run that bundle through the old code they would get an extra column for invalid data. But with the new code the same bundle would error when it sees invalid data.

Reply:
Unfortunately, HandleInvalid.error needs to be the default for old spark models to be backwards compatible. I made previously the change to just use HandleInvalid.Keep thinking it works for Spark models too, but it doesn't :( It's been a bit of a nightmare to get this 1he updates working.